packages/components/filters/filters-list/src/filters-list.stories.tsx

```diff
@@ -1,22 +1,162 @@
+  { label: 'Green', value: 'green', key: 'green', id: '7' },
+];
+
+const secondaryColorOptions = [
+  { label: 'Purple', value: 'purple', key: 'purple', id: '3' },
+  { label: 'Forest', value: 'forest', key: 'forest', id: '8' },
+  { label: 'Slate', value: 'slate', key: 'slate', id: '9' },
+  { label: 'Silver', value: 'silver', key: 'silver', id: '10' },
+];
+
+export const BasicExample: Story = () => {
+  const [primaryColorValue, setPrimaryColorValue] = useState<string[]>([]);
+  const [secondaryColorValue, setSecondaryColorValue] = useState<string[]>([]);
+
+  const clearPrimaryColorFilter = () => setPrimaryColorValue([]);
+  const clearSecondaryColorFilter = () => setSecondaryColorValue([]);
+
+  const getPrimaryColorTags = useCallback(() => {
+    return primaryColorValue.length > 0
+      ? primaryColorValue.map((value) => ({
+          key: value,
+          label: value.toUpperCase(),
+        }))
+      : [];
+  }, [primaryColorValue]);
+
+  const getSecondaryColorTags = useCallback(() => {
+    return secondaryColorValue.length > 0
+      ? secondaryColorValue.map((value) => ({
+          key: value,
+          label: value.toUpperCase(),
+        }))
+      : [];
+  }, [secondaryColorValue]);
+
+  const clearAllFilters = () => {
+    clearPrimaryColorFilter();
+    clearSecondaryColorFilter();
+  };
+
+  const filters = [
+    {
+      key: 'primaryColors',
+      label: 'Primary Colors',
+      groupKey: 'primaryColors',
+      getTags: getPrimaryColorTags,
+      onClearRequest: clearPrimaryColorFilter,
+    },
+    {
+      key: 'secondaryColors',
+      label: 'Secondary Colors',
+      groupKey: 'secondaryColors',
+      getTags: getSecondaryColorTags,
+      onClearRequest: clearSecondaryColorFilter,
+    },
+  ];
+
+  const filterConfigs = [
+    {
+      filterKey: 'primaryColors',
+      name: 'primaryColorsSelect',
+      options: primaryColorOptions,
+      value: primaryColorValue,
+      setValue: setPrimaryColorValue,
+      isMulti: false,
+    },
+    {
+      filterKey: 'secondaryColors',
+      name: 'secondaryColorsSelect',
+      options: secondaryColorOptions,
+      value: secondaryColorValue,
+      setValue: setSecondaryColorValue,
+      isMulti: true,
+    },
+  ];
+
+  const filterGroups = [
+    { key: 'secondaryColors', label: <div>SECONDARY COLORS</div> },
+    { key: 'primaryColors', label: <div>PRIMARY!!!</div> },
+  ];
 
-export const BasicExample: Story = {
-  args: {
-    label: 'A list of filters',
-  },
+  return (
+    <div
+      style={{
+        height: 500,
+        position: 'relative',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+      }}
+    >
+      <FiltersList
+        filters={filters}
+        onClearAllRequest={clearAllFilters}
+        filterGroups={filterGroups}
+      >
+        {filterConfigs.map(
+          ({ filterKey, name, options, value, setValue, isMulti = false }) => (
```

Comment:
I was actually hoping for some sort of component ensemble which would reduce the need for custom props. 
e.g. a  `<FilterList.ClearFiltersButton/>` component would allow us to ditch the `onClearRequest` on the `FiltersList`, cause the consuming app could just use the regular `onClick` handler of the `<FilterList.ClearFiltersButton/>`.

Reply:
`onClearAllRequest` is a required prop because the design spec requires a "clear all" button to be displayed in the `FiltersList`.  Making a `FilterList.ClearFiltersButton` component available and expecting it to be passed as a child of the `FiltersList` opens up the possibility of the consuming application not implementing the "clear all" button.

I think it makes a lot of sense to allow the consuming app to pass components a children/props when what they are defining is optional, but when something is required I think there is a lot to be said for making it as straightforward as possible to implement.

Reply:
> I think it makes a lot of sense to allow the consuming app to pass components a children/props when what they are defining is optional, but when something is required I think there is a lot to be said for making it as straightforward as possible to implement.

This is part of what I was saying to you earlier today @misama-ct about the risk of passing required props as children and having the consumer decide on it as though it was optional.